src/blueprint/from_config.rs

```diff
@@ -32,17 +32,29 @@ pub fn config_blueprint(config: &Config) -> Valid<Blueprint> {
   let schema = to_schema(config)?;
   let definitions = to_definitions(config, output_types, input_types)?;
   let server: Server = Server::try_from(config.server.clone())?;
-  let blueprint = Blueprint { schema, definitions, server };
+  let mut upstream = config.upstream.clone();
+  upstream.base_url = handle_base_url(upstream.base_url)?;
+  let blueprint = Blueprint { schema, definitions, server, upstream };
   let blueprint = apply_batching(blueprint);
   Ok(super::compress::compress(blueprint))
 }
 
+fn handle_base_url(base_url: Option<String>) -> Valid<Option<String>> {
```

Comment:
```suggestion
fn handle_base_url(base_url: Option<String>) -> Valid<()> {
```
This function doesn't need to return anything, just validate.